mkosi/gentoo.py

```diff
@@ -0,0 +1,549 @@
+            bins = []
+            usr = []
+            usrreg = "^[.]/(etc|usr|var)"
+            binreg = "^[.]/(sbin|bin|lib|lib64)"
+            devreg = "^[.]/dev"
+            runreg = "^[.]/run"
+
+            members = tfd.getmembers()
+            usrpath = root / "usr"
+
+            ignore = ["./var/run", "./var/lock", "./bin/awk"]
+            for dir in self.UNINSTALL_IGNORE:
+                ignore += [dir, f"./usr/{dir}"]
+
+            for tarinfo in members:
+                if tarinfo.name in ignore or re.match(devreg, tarinfo.name) or re.match(runreg, tarinfo.name):
+                    if "build-script" in ARG_DEBUG:
+                        print(f"{tarinfo.name} -> /dev/null")
+                    continue
+                elif (
+                    tarinfo.issym()
+                    and (os.path.dirname(tarinfo.name) == "./usr/bin")
+                    and (os.path.dirname(tarinfo.linkname) == "../../bin")
+                ):
+                    if "build-script" in ARG_DEBUG:
+                        print(f"{tarinfo.name} -> {tarinfo.linkname} -> /dev/null")
+                    continue
+                elif re.match(usrreg, tarinfo.name):
+                    usr.append(tarinfo)
+                elif re.match(binreg, tarinfo.name):
+                    bins.append(tarinfo)
+                else:
+                    print(f"{tarinfo.name} -> /dev/null")
+                    continue
+
+            try:
+                tfd.extractall(usrpath, bins, numeric_owner=True)
+                tfd.extractall(root, usr, numeric_owner=True)
+            except FileExistsError:
+                pass
+
+            copy_path(root.joinpath("usr/sbin"), root.joinpath("usr/bin"))
+
+    def set_profile(self, args: CommandLineArguments) -> None:
+        # be careful not to overwrite user's profile, (via skeleton)
+        if not self.profile_path.is_symlink():
+            MkosiPrinter.print_step(f"{args.distribution} setting Profile")
+            self.profile_path.symlink_to(self.portage_cfg["PORTDIR"] / self.arch_profile)
+
+        return
+
+    def set_default_repo(self) -> None:
+        eselect_repo_conf = self.portage_cfg_dir / "repos.conf"
+        eselect_repo_conf.mkdir(exist_ok=True)
+        eselect_repo_conf.joinpath("eselect-repo.conf").write_text(
+            dedent(
+                f"""\
+                    [gentoo]
+                    location = {self.portage_cfg["PORTDIR"]}
+                    sync-uri = https://anongit.gentoo.org/git/repo/gentoo.git
+                    sync-type = git
+                    sync-dept = 1
+                    """
+            )
+        )
+
+        return
+
+    def unmask_arch(self) -> None:
+        package_accept_keywords = self.portage_cfg_dir / "package.accept_keywords"
+        package_accept_keywords.mkdir(exist_ok=True)
+
+        package_accept_keywords.joinpath("mkosi").write_text(
+            dedent(
+                # homed is still ~ARCH
```

Comment:
That comment is a bit terse, as a non-Gentoo user I have no idea what it means. Mind to clarify?